qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java

```diff
@@ -0,0 +1,246 @@
+		dashboardPage.goToHomePage();
+	}
+
+	@And("a user clicks on the Manage Provider Schedules app")
+	public void loadAppointmentBlocksPage() {
+		appointmentBlocksPage = (AppointmentBlocksPage ) appointmentSchedulingPage.goToAppointmentBlock();
+	}
+
+	@And("the system loads Appointment Blocks page")
+	public void systemLoadsAppointmentBlocksPage() {
+		assertTrue(textExists("Manage Appointment Blocks"));
+	}
+
+	@And("a user clicks on an appointment block")
+	public void clickOnAppointmentBlock() throws InterruptedException {
+		appointmentBlocksPage.selectLocation(LOCATION);
+		appointmentBlocksPage.clickOnCurrentDay();
+		appointmentBlocksPage.clickOnAppointment();
+	}
+
+	@And("a user clicks on Edit appointment block")
+	public void loadEditAppointmentBlockSection() {
+		appointmentBlocksPage.clickOnEdit();
+	}
+
+	@And("the system loads Edit Appointment Block section")
+	public void systemLoadsEditAppointmentBlockSection() {
+		assertTrue(textExists("Edit Appointment Block"));
+	}
+
+	@And("a user edits the Appointment block")
+	public void editAppointmentBlock() {
+		appointmentBlocksPage.removeAppointment();
+		appointmentBlocksPage.enterService(SECOND_SERVICE_NAME);
+	}
+
+	@And("a user clicks on Save button")
+	public void saveAppointmentBlock() {
+		appointmentBlocksPage.clickOnSave();
+	}
+
+	@Then("the system saves the edited appointment block")
+	public void systemSavesEditedAppointmentBlock() {
+		assertTrue(appointmentBlocksPage.containsText(SECOND_SERVICE_NAME));
+	}
+
+	@And("a user clicks on Delete appointment block")
+	public void deleteAppointmentBlock() throws InterruptedException {
+		appointmentBlocksPage.clickOnDelete();
+		appointmentBlocksPage.clickOnConfirmDelete();
+		appointmentBlocksPage.waitForPage();
+	}
+
+	@Then("the system deletes appointment block from the appointment Blocks table")
+	public void systemDeletesAppointmentBlock() {
+		assertTrue(appointmentBlocksPage.containsText("Manage Appointment Blocks"));
+		dashboardPage.goToHomePage();
+	}
+
+	@And("a user clicks on Manage Appointments app")
+	public void loadFindPatientPage() {
+		findPatientPage = appointmentSchedulingPage.goToManageAppointments();
+	}
+
+	@And("user searches patient")
+	public void searchPatient() {
+		findPatientPage.enterPatient(testPatient.getName());
+	}
+
+	@And("user selects first patient")
+	public void selectFirstPatient() {
+		manageAppointmentsPage = (ManageAppointmentsPage) findPatientPage.clickOnFirstPatientAppointment();
+	}
+
+	@And("a user searches appointment request")
+	public void searchAppointmentRequest() {
+		manageAppointmentsPage.clickOnViewAllTypes();
+		manageAppointmentsPage.clickOnService(FIRST_SERVICE_NAME);
+		manageAppointmentsPage.searchAppointment();
+		manageAppointmentsPage.clickAppointment();
+	}
+
+	@And("a user clicks on book appointment icon")
+	public void clickOnBookAppointment() {
+		manageAppointmentsPage.clickOnBookAppointment();
+	}
+
+	@And("a user clicks the Save button")
```

Comment:
```suggestion
	@And("a user clicks on  Save button")
```

Reply:
@Irenyak1  Well done on this, Left few simply changes to consider, then this is good to be merged.